### DISCUSSION TITLE
Public API or IFrame for validator.schema.org

### DISCUSSION THREAD

**Discussion by rdutta-kareo** [schemaorg/schemaorg#3261] [General]
I want to run a public API for https://validator.schema.org/ on multiple of our Clients websites, to test markup. We want to obtain the results for our React Web project, and display in a Table. 

Is there an general API to run this, numerous times? 
I am getting 429 Too many requests for calling the API 50 times in an hour.  Should I just avoid calling the API entirely, and use the github project locally? At the very least, can you open the website for Iframe, so it can be used in other websites?

POST: request x-www-form-urlencoded
https://validator.schema.org/validate

**Comment by danbri** [comment]
Do not try to scrape validator.schema.org - it is a service run and hosted
by Google as a contribution to the project. This Github doesn’t include it.
There are no resources or plans for Google to make an API. Rather we should
work on parser interop with related tools.

**Reply by rdutta-kareo** [reply]
ok thank you, is there any method to acquire the data multiple times without 429 (overrequest) error? is anyone able to open the website to iframe? Any secondary tools you can recommend?

**Comment by danbri** [comment]
It's basically a Schema.org -themed version of the old Google Structured Data Testing Tool, and as a Google service has a lot of safeguards against scraping. I will close this issue as I don't think there's an easy answer for your specific question. If you enjoy wrestling with JS code you might find https://github.com/google/schemarama/ useful/interesting, as a work in progress which packages up JSON-LD, RDFa, Microdata parsers alongside SHACL and ShEx validation. It isn't a drop-in solution for you though.

**Comment by danbri** [comment]
Actually I just noticed this is a discussion not an Open Issue so there's nothing to close. How nice :) We have way too many open issues...

**Comment by philbarker** [comment]
If there is a SHACL description of the requirements and if you can get the data as just JSON-LD (not in an HTML page), there is the EU DIGIT Interoperability Test Bed service that lets you upload data and SHACL to validate the one against the other: https://www.itb.ec.europa.eu/shacl/any/upload

It's relative straightforward to set up the testbed to run locally in docker, https://www.itb.ec.europa.eu/docs/guides/latest/installingTheTestBed/index.html

That service is nothing to do with me, I just found it useful.

HTH Phil

On 13/02/2023 18:30, Dan Brickley wrote:

> It's basically a Schema.org -themed version of the old Google Structured Data Testing Tool, and as a Google service has a lot of safeguards against scraping. I will close this issue as I don't think there's an easy answer for your specific question. If you enjoy wrestling with JS code you might find https://github.com/google/schemarama/ useful/interesting, as a work in progress which packages up JSON-LD, RDFa, Microdata parsers alongside SHACL and ShEx validation. It isn't a drop-in solution for you though.
>
> —
> Reply to this email directly, [view it on GitHub](https://github.com/schemaorg/schemaorg/discussions/3261#discussioncomment-4961781), or [unsubscribe](https://github.com/notifications/unsubscribe-auth/AAFAU77MOWOG5SVC3UXRRJTWXJ4VNANCNFSM6AAAAAAUZ7NZOI).
> You are receiving this because you are subscribed to this thread.Message ID: ***@***.***>

--

[Phil Barker](http://people.pjjk.net/phil), http://people.pjjk.net/phil (he/him).
[CETIS LLP](https://www.cetis.org.uk): a cooperative consultancy for innovation in education technology.
[PJJK Limited](https://www.pjjk.co.uk): technology to enhance learning; information systems for education.

CETIS is a co-operative limited liability partnership, registered in England number OC399090
PJJK Limited is registered in Scotland as a private limited company, number SC569282.
